internal/monitors/utilization/utilization_windows.go

```diff
@@ -0,0 +1,126 @@
+// +build windows
+
+package utilization
+
+import (
+	"context"
+	"time"
+
+	"github.com/shirou/gopsutil/mem"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/signalfx-agent/internal/monitors/telegraf/common/accumulator"
+	"github.com/signalfx/signalfx-agent/internal/monitors/telegraf/common/emitter/batchemitter"
+	"github.com/signalfx/signalfx-agent/internal/monitors/telegraf/common/measurement"
+	"github.com/signalfx/signalfx-agent/internal/monitors/telegraf/monitors/winperfcounters"
+	"github.com/signalfx/signalfx-agent/internal/monitors/utilization/perfcounter"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+)
+
+// emitMemoryUtilizaion collects and emits memory metrics via gopstuil.
+// We do this because we can't get all of the memory metrics via perf counter
+// on all supported win versions.
+func (m *Monitor) emitMemoryUtilization() {
+	// mem.VirtualMemory is a gopsutil function
+	memInfo, _ := mem.VirtualMemory()
+	dimensions := map[string]string{"plugin": monitorType}
+	// perfcounter: ""; perfcounter reporter: "memory.available_mbytes"; collectd: "memory.free";
+	m.Output.SendDatapoint(datapoint.New("memory.free", dimensions, datapoint.NewIntValue(int64(memInfo.Available)), datapoint.Gauge, time.Now()))
+	// perfcounter: ""; perfcounter reporter: "signalfx.usedmemory"; collectd: "memory.used";
+	m.Output.SendDatapoint(datapoint.New("memory.used", dimensions, datapoint.NewIntValue(int64(memInfo.Used)), datapoint.Gauge, time.Now()))
+	// perfcounter: ""; perfcounter reporter: ""; collectd: "memory.utilization"
+	util := (float64(memInfo.Used) / float64(memInfo.Total)) * 100
+	m.Output.SendDatapoint(datapoint.New("memory.utilization", dimensions, datapoint.NewFloatValue(util), datapoint.Gauge, time.Now()))
+}
+
+func (m *Monitor) getPerfCounters() ([]winperfcounters.PerfCounterObj, func([]*measurement.Measurement)) {
+	memory := perfcounter.Memory()
+	processor := perfcounter.Processor()
+	logicalDisk := perfcounter.LogicalDisk()
+	physicalDisk := perfcounter.PhysicalDisk()
+	networkIntf := perfcounter.NetworkInterface()
+	pagefile := perfcounter.PageFile()
+	processMeasurements := func(measurements []*measurement.Measurement) {
+		for _, measurement := range measurements {
+			var err []error
+			switch measurement.Measurement {
```

Comment:
This is kind of a code smell having so much repetition of the same set of 6 components three times in this function, esp. this switch statement.  You could eliminate the switch statement with some interface use but it doesn't seem easy given how the code is structured.